fio/fio_run

```diff
@@ -254,22 +254,50 @@ report_info()
 	echo $1 >> /tmp/log
 }
 
+#
+# Averaging function for json formatted bw and iops
+#
 obtain_avg()
 {
-	value=`grep "${1}" fio-result.txt | grep max | cut -d'=' -f ${2} | grep -v percentiles | cut -d',' -f 1`
-	nitems=0
-	calc="scale=2;("
-	field_separ=""
-	for val in $value; do
-		calc="${calc}${field_separ}$val"
-		let "nitems=$nitems+1"
-		field_separ="+"
-	done
-	calc="${calc})/${nitems}"
-	rtval=`echo $calc | bc`
-	echo $rtval
+        FIELD=${1}
+        value=`cat fio-results.json | jq '.. | select(type == "object" and has('\"$FIELD\"')).'\"$FIELD\"''`
```

Comment:
jq can take a file as a the second argument, remove pipe from `cat`

Reply:
done